src/amb_kf.c

```diff
@@ -208,18 +314,35 @@ static void make_residual_measurements(nkf_t *kf, double *measurements, double *
   }
 }
 
+/** The temporal update step of the KF.
+ * To be really strict, since changes are equally likely on all channels,
+ * This should be += (I + 1 * 1^T)*var instead of I*var, but it's
+ * unlikely to be significant.
+ *
+ * \param kf The KF to be updated.
+ */
 static void diffuse_state(nkf_t *kf)
 {
+  double cov[kf->state_dim * kf->state_dim];
+  matrix_reconstruct_udu(kf->state_dim, kf->state_cov_U, kf->state_cov_D, cov);
```

Comment:
We used to just update the KF cov by doing D += var. In some experimentation, the covariances are actually quite different doing it that way as opposed to doing UDU += var, so I switched it back.